jgrapht-core/src/main/java/org/jgrapht/graph/SynchronizedGraphParams.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * (C) Copyright 2018-2018, by CHEN Kui and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.graph;
+
+/**
+ * The SynchronizedGraph Params describes the properties of a synchronized graph being created such
+ * as whether it uses cache and fair mode.
+ *
+ * @author CHEN Kui
+ * @since Mar 9, 2018
+ */
+public class SynchronizedGraphParams
```

Comment:
I would suggest to move this class inside the `AsSynchronizedGraph` class as public static. No other class will need this. 

Moreover this class uses in a rather unconventional way the builder pattern. I would suggest to add standard getters and setters and use the Builder Pattern in a standard way.

Reply:
@d-michail do you think this should be static inner class AsSynchronizedGraph.Builder, or should it go in the existing GraphTypeBuilder?  (I noticed that you don't allow an AsUnmodifiableGraph to be built there, even though GraphType has an asUnmodifiable method, but I think that's because building an unmodifiable graph is kind of pointless since then it will always be empty.)

Another question:  do you think AsSynchronizedGraph belongs in its current location, or should we start a new package org.jgrapht.graph.concurrent?

Reply:
I was only referring to the class `SynchronizedGraphParams` which currently behaves both like a POJO and a builder. 

I think I would keep `GraphTypeBuilder` separate for now, maybe add a method `isSynchronized` in the `GraphType` just like unmodifiable. The As... classes are more like utilities which provide a different "view" of the graph.

Regarding the 'concurrent' package, since it would only contain on class, I am not sure.

Reply:
OK, let's keep the builder separate for now, and leave any modification to GraphType for a later pull request (not this one).

For the concurrent package, I was thinking we would eventually want to add additional implementations such as snapshot-capable.  So instead of having to move this one later to join them, we could start it off in the right place.

Reply:
@Yimismi can you implement Dimitrios' suggestion for replacing SynchronizedGraphParams with the builder pattern (via a static inner class AsSynchronizedGraph.Builder)?

Also, let's go ahead now with moving AsSynchronizedGraph to a new package org.jgrapht.graph.concurrent so that we can avoid having to move it later.

Reply:
I'm sorry that I missed this message.